src/manifolds/Tucker.jl

```diff
@@ -0,0 +1,741 @@
+matrices of `v` are in the orthogonal complement of the HOSVD factors of `p`.
+"""
+function check_vector(
+    M::Tucker{N,R,D}, p::TuckerPoint{T,D}, v::TuckerTVector
+) where {N,R,T,D}
+    s = "The tangent vector $(v) is not a tangent vector to $(p) on $(M), "
+    if size(p.hosvd.core) ≠ size(v.Ċ) || any(size.(v.U̇) .≠ size.(p.hosvd.U))
+        return DomainError(
+            size(v.Ċ), s * "since the array dimensons of $(p) and $(v)" * "do not agree."
+        )
+    end
+    for (U, U̇) in zip(p.hosvd.U, v.U̇)
+        if norm(U' * U̇) ≥ √eps(eltype(U)) * √length(U)
+            return DomainError(
+                norm(U' * U̇),
+                s *
+                "since the columns of x.hosvd.U are not" *
+                "orthogonal to those of v.U̇.",
+            )
+        end
+    end
+    return nothing
+end
+
+"""
+    Base.convert(::Type{Matrix{T}}, basis :: CachedBasis{𝔽,DefaultOrthonormalBasis{𝔽, TangentSpaceType},HOSVDBasis{T, D}}) where {𝔽, T, D}
+    Base.convert(::Type{Matrix}, basis :: CachedBasis{𝔽,DefaultOrthonormalBasis{𝔽, TangentSpaceType},HOSVDBasis{T, D}}) where {𝔽, T, D}
+
+Convert a HOSVD basis to a matrix whose columns are the vectorisations of the basis vectors.
+"""
+function Base.convert(::Type{Matrix{T}}, ℬ::CachedHOSVDBasis{𝔽,T,D}) where {𝔽,T,D}
+    𝔄 = ℬ.data.point
+    r⃗ = size(𝔄.hosvd.core)
+    n⃗ = size(𝔄)
+    ℳ = Tucker(n⃗, r⃗)
+
+    J = Matrix{T}(undef, prod(n⃗), manifold_dimension(ℳ))
+    # compute all possible ∂𝔄╱∂ℭ (in one go is quicker than one vector at a time)
+    J[:, 1:prod(r⃗)] = ⊗ᴿ(𝔄.hosvd.U...)
+    # compute all possible ∂𝔄╱∂U[d] for d = 1,...,D
+    function fill_column!(i, vᵢ)
+        Jᵢ_tensor = reshape(view(J, :, i), n⃗) # changes to this apply to J as well
+        return embed!(ℳ, Jᵢ_tensor, 𝔄, vᵢ)
+    end
+    foreach(fill_column!, ℳ, 𝔄, ℬ, (prod(r⃗) + 1):manifold_dimension(ℳ))
+    return J
+end
+function Base.convert(::Type{Matrix}, basis::CachedHOSVDBasis{𝔽,T,D}) where {𝔽,T,D}
+    return convert(Matrix{T}, basis)
+end
+
+Base.copy(x::TuckerTVector) = TuckerTVector(copy(x.Ċ), map(copy, x.U̇))
+
+function Base.copyto!(q::TuckerPoint, p::TuckerPoint)
+    for d in 1:ndims(q)
+        copyto!(q.hosvd.U[d], p.hosvd.U[d])
+        copyto!(q.hosvd.σ[d], p.hosvd.σ[d])
+    end
+    copyto!(q.hosvd.core, p.hosvd.core)
+    return q
+end
+function Base.copyto!(y::TuckerTVector, x::TuckerTVector)
+    for d in 1:ndims(y.Ċ)
+        copyto!(y.U̇[d], x.U̇[d])
+    end
+    copyto!(y.Ċ, x.Ċ)
+    return y
+end
+
+@doc raw"""
+    embed(::Tucker, A :: TuckerPoint)
+
+Convert a point `A` on the Tucker manifold to a full tensor, represented as an
+$N_1 \times \dots \times N_D$-array.
+
+    embed(::Tucker, A::TuckerPoint, X::TuckerTVector)
+
+Convert a tangent vector `X` with base point `A` on the Tucker manifold to a full tensor,
+epresented as an $N_1 \times \dots \times N_D$-array.
+"""
+embed(::Tucker, ::Any, ::TuckerPoint)
+
+function embed!(::Tucker, q, p::TuckerPoint)
```

Comment:
This might not be necessary if the allocation is correct? Also `Tucker`should maybe be an `AbstractEmbeddedManifold` where `get_embedding` is then defined to return the `Euclidean` of corresponding size (compare e.g. to Sphere or Hyperbolic); oh I noticed, for FixedRank it might also be nice, maybe.

Reply:
A `TuckerPoint` is stored in factorised form. This is the only function where the factorisation is evaluated explicitly. So the default `embed!`  implementation won't work.

Reply:
Sure, you still have to implement `embed!`. But it would be more precisely typed. I am not 100% sure it would bring any actual benefits (see Fixedrank, where it seems missing but nobody noticed yet).